xapian-core/cluster/round_robin.cc

```diff
@@ -0,0 +1,63 @@
+/** @file round_robin.cc
+ *  @brief Round Robin clustering API
+ */
+/* Copyright (C) 2016 Richhiey Thomas
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+#include <config.h>
+
+#include "xapian/cluster.h"
+
+#include "debuglog.h"
+
+using namespace Xapian;
+using namespace std;
+
+string
+RoundRobin::get_description() const
+{
+    return "RoundRobin()";
+}
+
+ClusterSet
+RoundRobin::cluster(const MSet &mset)
+{
+    LOGCALL(API, ClusterSet, "RoundRobin::cluster", mset);
+    TermListGroup tlg(mset);
+    ClusterSet cset;
+    vector<Point> points;
+
+    for (MSetIterator it = mset.begin(); it != mset.end(); ++it) {
+	Point p;
+	p.initialize(tlg, it.get_document());
+	points.push_back(p);
```

Comment:
Is it useful to initialise a `Point` more than once?

If not, perhaps `initialize()` should be turned into a constructor instead.  Then you can construct and push in one go, which would help to avoid copying of objects:

```
points.push_back(Point(tlg, it.get_document());
```

(If we keep the `initialize()` method, it probably should be spelled `initialise()` instead - we've so far used British English rather than US English spellings in the API in the few cases where it makes a difference).